commerce_coordinator/apps/commercetools/clients.py

```diff
@@ -369,6 +369,18 @@ def get_payment_by_transaction_interaction_id(self, interaction_id: str) -> CTPa
         logger.info(f"[CommercetoolsAPIClient] - Attempting to find payment with interaction ID {interaction_id}")
         return self.base_client.payments.query(where=f'transactions(interactionId="{interaction_id}")').results[0]
 
+    def get_product_by_program_id(self, program_id: str) -> Optional[CTProductVariant]:
+        """
+        Fetches a program from Commercetools.
+        Args:
+            program_id: The ID of the program (bundle) to fetch.
+        Returns:
+            CTProductVariant if found, None otherwise.
+        """
+        results = self.base_client.product_projections.search(False, filter=f'key:"{program_id}"').results
```

Comment:
Is this method more performant than the previously used get_by_id?

Reply:
yes two reasons for using this.

- As per SDK, this is the high performance endpoint.
- `get_by_key` returns exception when product is not found. But search simply returns empty array if no matching product found